pkg/controller/atlasproject/encryption_at_rest_test.go

```diff
@@ -5,13 +5,295 @@ import (
+				Data: secretData,
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Secret",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "aws-secret",
+					Namespace: "test",
+				},
+			},
+		}...).Build()
+
+		service := &workflow.Context{}
+
+		encRest := &mdbv1.EncryptionAtRest{
+			AwsKms: mdbv1.AwsKms{
+				Enabled: toptr.MakePtr(true),
+				SecretRef: common.ResourceRefNamespaced{
+					Name:      "aws-secret",
+					Namespace: "test",
+				},
+			},
+		}
+
+		err := readEncryptionAtRestSecrets(kk, service, encRest, "test")
+		assert.Nil(t, err)
+
+		assert.Equal(t, string(secretData["CustomerMasterKeyID"]), encRest.AwsKms.CustomerMasterKeyID)
+		assert.Equal(t, string(secretData["Region"]), encRest.AwsKms.Region)
+		assert.Equal(t, string(secretData["RoleID"]), encRest.AwsKms.RoleID)
+	})
+
+	t.Run("AWS with correct secret data (fallback namespace)", func(t *testing.T) {
+		secretData := map[string][]byte{
+			"CustomerMasterKeyID": []byte("testCustomerMasterKeyID"),
+			"Region":              []byte("testRegion"),
+			"RoleID":              []byte("testRoleID"),
+		}
+
+		kk := fake.NewClientBuilder().WithRuntimeObjects([]runtime.Object{
+			&v1.Secret{
+				Data: secretData,
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Secret",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "aws-secret",
+					Namespace: "test-fallback-ns",
+				},
+			},
+		}...).Build()
+
+		service := &workflow.Context{}
+
+		encRest := &mdbv1.EncryptionAtRest{
+			AwsKms: mdbv1.AwsKms{
+				Enabled: toptr.MakePtr(true),
+				SecretRef: common.ResourceRefNamespaced{
+					Name: "aws-secret",
+				},
+			},
+		}
+
+		err := readEncryptionAtRestSecrets(kk, service, encRest, "test-fallback-ns")
+		assert.Nil(t, err)
+
+		assert.Equal(t, string(secretData["CustomerMasterKeyID"]), encRest.AwsKms.CustomerMasterKeyID)
+		assert.Equal(t, string(secretData["Region"]), encRest.AwsKms.Region)
+		assert.Equal(t, string(secretData["RoleID"]), encRest.AwsKms.RoleID)
+	})
+
+	t.Run("AWS with missing fields", func(t *testing.T) {
+		secretData := map[string][]byte{
+			"AccessKeyID":         []byte("testKeyID"),
+			"SecretAccessKey":     []byte("testSecretAccesssKey"),
+			"CustomerMasterKeyID": []byte("testCustomerMasterKeyID"),
+		}
+
+		kk := fake.NewClientBuilder().WithRuntimeObjects([]runtime.Object{
+			&v1.Secret{
+				Data: secretData,
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Secret",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "aws-secret",
+					Namespace: "test",
+				},
+			},
+		}...).Build()
+
+		service := &workflow.Context{}
+
+		encRest := &mdbv1.EncryptionAtRest{
+			AwsKms: mdbv1.AwsKms{
+				Enabled: toptr.MakePtr(true),
+				SecretRef: common.ResourceRefNamespaced{
+					Name:      "aws-secret",
+					Namespace: "test",
+				},
+			},
+		}
+
+		err := readEncryptionAtRestSecrets(kk, service, encRest, "test")
+		assert.NotNil(t, err)
```

Comment:
Shouldn't we check for a particular error or text within the error?

Reply:
We can, but in this case it is just to check that all fields were read correctly and there are no problems with a secret data